### ISSUE TITLE
can't restart server

### ISSUE THREAD

**Issue by xchoice** [itzg/docker-minecraft-server#522]
Modpack: Manufactio
url: https://www.curseforge.com/minecraft/modpacks/manufactio/files/2943928

docker compose:
```
version: "3.7"
services:
  mc:
    image: itzg/minecraft-server
    ports:
      - 25565:25565
    volumes:
      - /srv/minecraft/Manufactio:/data
    environment:
      EULA: "TRUE"
      MEMORY: 8G
      TYPE: CURSE_INSTANCE
      VERSION: 1.12.2
      CURSE_INSTANCE_JSON: minecraftinstance.json
```

error:
[init] Setting initial memory to 8G and max to 8G
2020-05-08T17:41:06.794Z        INFO    mc-server-runner.cfsync Preparing Twitch/Curse instance {"name": "Manufactio"}
2020-05-08T17:41:06.795Z        INFO    mc-server-runner.cfsync Preparing forge server
2020-05-08T17:41:06.795Z        INFO    mc-server-runner.cfsync Preparing libraries for forge server
2020-05-08T17:41:06.861Z        INFO    mc-server-runner.cfsync Preparing mods
2020-05-08T17:41:06.862Z        FATAL   mc-server-runner        Failed to prepare instance      {"error": "failed to prepare mods: failed to remove old mod file: remove mods/memory_repo: directory not empty"}

For the setup, I downloaded the pack from twitch app, copied the files to the attached volume, and started the server with docker-compose up -d. The server starts without problem.

My problem is when I stop the server, and try to bring it up again.
Sometimes instead of "mods/memory_repo" it complains about "mods/1.12.2". If I remove both folders, the server starts again without problem.

**Comment by itzg** [comment]
I'll see if I can recreate the issue and fix it. The `TYPE: CURSE_INSTANCE` is a beta feature, so I probably have some scenarios like this to fix.

**Comment by xchoice** [comment]
With all sincerity, I have no clue what type should I use, generic_pack forge curse_instance, quiet confusing >.<

**Comment by itzg** [comment]
Supporting and keeping up with all the types is daunting too :) 

If that modpack provided server files, I would have recommended the [TYPE=CURSEFORGE](https://github.com/itzg/docker-minecraft-server#running-a-server-with-a-feed-the-beast-ftb--curseforge-modpack) approach. Since it doesn't, you're on the right track with using the new `CURSE_INSTANCE` type.

**Comment by itzg** [comment]
Fixed in 5c0a3a64abf30a0c26dcc9dc849fdaa4d87cbab0 and available on Docker Hub. Let me know if you still see issues after pulling the latest image.
